config/printer-tronxy-X5SA-and-X5SAPro.cfg

```diff
@@ -0,0 +1,228 @@
+# Be sure to set the position_max to match your machines maximum dimension per axis.
+[stepper_x]
+step_pin: PE5
+dir_pin: !PE6 # X5SA
+enable_pin: !PC13
+microsteps: 16
+rotation_distance: 20
+endstop_pin: !PG10
+position_endstop: -1
+position_min: -1
+position_max: 330 # This value is kept small for safety. Change to 330, 400, or 500 for the appropriate X5SA model.
+homing_speed: 50
+homing_retract_dist: 10
+second_homing_speed: 10.0
+
+[stepper_y]
+step_pin: PE2
+dir_pin: !PE3 # X5SA
+enable_pin: !PE4
+microsteps: 16
+rotation_distance: 20
+endstop_pin: !PA12
+position_endstop: 0
+position_max: 330 # This value is kept small for safety. Change to 330, 400, or 500 for the appropriate X5SA model.
+homing_retract_dist: 10
+homing_speed: 50.0
+second_homing_speed: 10.0
+
+[stepper_z]
+step_pin: PB9
+dir_pin: PE0 # X5SA
+enable_pin: !PE1
+microsteps: 16
+rotation_distance: 4
+# endstop_pin must be set to either the actual pin value for a true microswitch, or set to probe:z_virtual_endstop if using a bed probe
+endstop_pin: probe:z_virtual_endstop
+# Other endstop headers are Z- (!PA14) and Z+ (!PA13)
+# Uncomment position_endstop and set a Z home value if using a true endstop switch
+#position_endstop: 0
+position_max: 400 # Change to 400 or 500 for the appropriate X5SA model.
+position_min: -2
+
+
+# Tronxy used two different extruder models over the years. A standard Creality
+# style, and a Titan style. These extruders run in opposite directions and require
+# different values for rotation distance. The dir_pin and rotation_distance are segregated below.
+# Uncomment the appropriate set of values. These values should get you close, however
+# it is always best to calibrate your rotation distance on your machine.
+
+[extruder]
+# The next three lines are for a Standard Extruder.
+#dir_pin: PB5 # Standard Extruder stepper direction
+# microsteps: 16 # Standard Extruder
+#rotation_distance: 35.520 # Standard Extruder
+# The next four lines are for a Titan Extruder
+dir_pin: !PB5 # Titan Extruder Clone stepper direction
+microsteps: 32 # Titan Extruder Clone Microsteps
+rotation_distance: 22.478 # Titan Extruder Clone Rotation Distance
+gear_ratio: 66:22 # Titan Extruder Clone Gear Ratio
+step_pin: PB4
+enable_pin: !PB8
+nozzle_diameter: 0.400
+filament_diameter: 1.750
+heater_pin: PG12
+sensor_type: ATC Semitec 104GT-2
+sensor_pin: PA1
+control: pid
+pid_Kp: 18.831
+pid_Ki: 0.821
+pid_Kd: 108.044
+min_temp: 0
+max_temp: 265
+max_extrude_only_distance: 300
+
+[heater_bed]
+heater_pin: PG11
+sensor_type: EPCOS 100K B57560G104F
+sensor_pin: PA0
+control: pid
+min_temp: 0
+max_temp: 130
+pid_Kp: 73.932
+pid_Ki: 1.521
+pid_Kd: 898.279
+
+[heater_fan hotend_fan]
+pin: PG14
+fan_speed: 1
+
+[fan]
+pin: PG13
+max_power: 1
+
+[controller_fan drivers_fan]
+pin: PD6
+
+[filament_switch_sensor sentinel]
+pause_on_runout: True
+runout_gcode:
+  M25
+switch_pin: !PA15
+
+[output_pin beeper]
+pin: PB0
```

Comment:
For my printer with the V6 Chitu board having this as a pure digital output pin doesn't work, as the board has a piezo buzzer which needs some PWM to get it to make an appropriate audible sound.
Unsure of how to do PWM but still have it as an On/Off type of signal (monotone).

I currently have the following in my config (setting the value to something like 1600 sounds 'good enough' for me, 0 to turn it off again).
[output_pin beeper]
pin: PB0
pwm: True ; Chitu V6 uses piezo buzzer which requires PWM
hardware_pwm: True
cycle_time: 0.001 ; PWM frequency : 0.001 = 1ms will give a base tone of 1kHz
scale: 4000 ; PWM parameter will be in the range of (0-4000 Hz).

Reply:
I never really cared about having the beeper active, so I didnt even consider this. I'll test it on the 191017 board I have here.